pyuvdata/mwa_corr_fits.py

```diff
@@ -69,9 +69,82 @@ def correct_cable_length(self, cable_lens):
         self.data_array *= np.exp(-1j * 2 * np.pi * cable_len_diffs / const.c.to('m/s').value
                                   * self.freq_array.reshape(1, self.Nfreqs))[:, :, None]
 
+    def flag_init(self, edge_width=80e3, start_flag=4.0, end_flag=6.0,
+                  flag_dc_offset=True):
+        """
+        Do routine flagging of the edges, beginning and end of obs, as well as
+        the center fine channel of each coarse channel.
+
+        Parameters
+        ----------
+        edge_width: float
+            The width to flag on the edge of each coarse channel, in hz. Set to
+            0 for no edge flagging.
+        start_flag: float
+            The number of seconds to flag at the beginning of the observation.
+            Set to 0 for no flagging.
+        end_flag: floats
+            The number of seconds to flag at the end of the observation. Set to
+            0 for no flagging.
+        flag_dc_offset: bool
+            Set to True to flag the center fine channel of each coarse channel.
+
+        Raises
+        ------
+        ValueError
+            If edge_width is not an integer multiple of the channel_width of the data (0 also acceptable).
+            If start_flag is not an integer multiple of the integration time (0 also acceptable).
+            If end_flag is not an integer multiple of the integration time (0 also acceptable).
+        """
+        if (edge_width % self.channel_width) > 0:
+            raise ValueError("The edge_width must be an integer multiple of the"
+                             "channel_width of the data or zero.")
+        if (start_flag % self.integration_time) > 0:
+            raise ValueError("The start_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+        if (end_flag % self.integration_time) > 0:
+            raise ValueError("The end_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+
+        num_ch_flag = int(edge_width / self.channel_width)
+        num_start_flag = int(start_flag / self.integration_time)
+        num_end_flag = int(end_flag / self.integration_time)
+
+        # Constant from the DSP architecture
+        num_coarse_chan = 24
+        num_fine_chan = len(self.Nfreqs) / num_coarse_chan
```

Comment:
Need to be careful here, we can pass only some of the coarse band files to read_mwa_corr_fits and we still want this to work.

Reply:
I'd suggest using the fact that we know the bandwidth of the coarse and fine channels.

Reply:
It turns out this number is computed during read_mwa_corr_fits using the header information in the box file, so I have opted to just pass that in from there in the most recent push.